userinterface/Services/FrameTimerService.cs

```diff
@@ -0,0 +1,84 @@
+using Avalonia.Threading;
+using System;
+using System.Diagnostics;
+
+namespace userinterface.Services
+{
+    // Service for monitoring UI thread blocking during performance-critical operations.
+    // Usage:
+    // - Call StartMonitoring("context") before performance-critical operations  
+    // - Call StopMonitoring("context") after completion
+    // - Use MonitorOperation("name", action) for automatic monitoring
+    // 
+    // To re-enable debug logging, uncomment Debug.WriteLine calls and add using System.Diagnostics;
```

Comment:
This class is great. But is it actually useful without the logging? Could instead take an ILogger that can be bound to either do nothing or to write to cmd etc.